safe/gui/tools/shake_grid/shake_grid.py

```diff
@@ -821,6 +835,85 @@ def set_contour_properties(self, input_file):
+
+        Get the MMI value on each cities/places from the generated tiff files.
+        :param place_layer: Point layer generated from distance calculation
+        :type place_layer: QgsVectorLayer
+        :param raster_mmi_path: Path to converted mmi raster.
+        :type raster_mmi_path: str
+        :return: Modified place layer containing MMI values
+        :rtype: QgsVectorLayer
+        """
+        shake_raster = QgsRasterLayer(raster_mmi_path, 'shake_raster')
+        shake_provider = shake_raster.dataProvider()
+
+        place_fields = place_layer.dataProvider().fields()
+        mmi_field_index = place_fields.indexFromName('mmi')
+
+        place_layer.startEditing()
+        for feature in place_layer.getFeatures():
+            fid = feature.id()
+            point = feature.geometry().asPoint()
+            mmi_identify = shake_provider.identify(
+                point,
+                QgsRaster.IdentifyFormatValue
+            )
+            mmi = mmi_identify.results()[1]
+            place_layer.changeAttributeValue(fid, mmi_field_index, mmi)
+        place_layer.commitChanges()
+        # save the memory place layer
+        data_store = Folder(self.output_dir)
+        data_store.default_vector_format = 'geojson'
+        data_store.add_layer(
+            place_layer,
+            '%s-nearby-places' % self.output_basename
+        )
+        return place_layer
+
+    def generate_locality_info(self, raster_mmi_path):
+        """Generate information related to the locality of the hazard.
+
+        :param raster_mmi_path: path to converted raster mmi
+        :type raster_mmi_path: str
+        """
+
+        epicenter = QgsPoint(self.longitude, self.latitude)
+        place_layer = self.place_layer
+
+        measured_place_layer = get_direction_distance(epicenter, place_layer)
+        mmi_point_layer = self.mmi_point_sampling(
+            measured_place_layer,
+            raster_mmi_path
+        )
+        places = []
+        for feature in mmi_point_layer.getFeatures():
+            place = {
+                'name': feature[self.name_field],
+                'population': feature[self.population_field],
+                'distance': feature['distance'],
+                'bearing_to': feature['bearing_to'],
+                'dir_to': feature['dir_to'],
+                'mmi_values': feature['mmi']
+            }
+            places.append(place)
+        # sort the place by mmi and by population
+        sorted_places = sorted(
+            places,
+            key=itemgetter('mmi_values', 'population'),
+            reverse=True
+        )
+        nearest_place = sorted_places[0]
+        # combine locality text
+        city_name = nearest_place['name']
+        city_distance = nearest_place['distance']
+        city_bearing = nearest_place['bearing_to']
+        city_direction = nearest_place['dir_to']
+        self.locality = 'Located ' + str(math.floor(city_distance / 1000)) + \
```

Comment:
Put it in `tr`. And use new string formatter to make it easy to translate.
e.g. 
```python
tr(
            'In this wizard step: {step_name}, you will be able to '
            'set a field that corresponded with a InaSAFE field '
            'concept.').format(step_name=self.step_name)

```